### PR TITLE
config refactor

### PR DESCRIPTION
**Pull request by zimny**
This pull request tired to address concerns in #14 and #8

### CODE REVIEW

File: docs/config/README.md [hunk 1]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
-The `Default` account may represent a developer personal account where they can test all Mira deployments. It uses the profile `my-user` defined in the `~/.aws` directory.
 
-## CI/CD
-`Staging` and `Production` accounts are defined in the `cicd.accounts` value. The CodePipeline then has `StagingDeploy` and `ProductionDeploy` stages which deploy your app into the accounts `2222222222222` and `4444444444444` respectively.
+## Accounts
+The accounts section of the config file is an array of account objects that can be used to deploy apps and run cicd. The accounts also represent environments.
+We define three accounts in the sample config file above: `staging`, `production`, `cicd`, each with different account numbers.
 
-AWS CodeBuild runs using the `infra/buildspec.yaml` file, where the directory is relative to the project's root path.
+The `staging` account may represent a developer personal account where they can test all Mira deployments. It uses the default profile defined in the `~/.aws` directory or the one passed as `--profile` parameter in the CLI command.
```
Comment (nigelhanlon) [inline docs/config/README.md:93]:
```suggestion
The `staging` account may represent a developer's personal account where they can test all Mira deployments. It uses the default profile defined in the `~/.aws` directory or the one passed in as `--profile` parameter in the CLI command.
```

File: docs/config/README.md [hunk 2]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
 
-The pipeline runs on the branch `feature/feature-xyz`.
 
+## CI/CD
+`cicd` section specifies stages of the deployment pipeline and other properties required for CI pipeline to work.
```
Comment (nigelhanlon) [inline docs/config/README.md:97]:
```suggestion
The `cicd` section specifies the various stages of the deployment pipeline and other properties required for the CI pipeline to work.
```

File: docs/config/README.md [hunk 3]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
+* `target` - Name of the account where CI pipeline will be deployed.
```
Comment (nigelhanlon) [inline docs/config/README.md:98]:
```suggestion
* `target` - Name of the account where the CI pipeline will be deployed.
```

File: docs/config/README.md [hunk 4]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
+    __Caution:__ Make sure to not include any whitespace characters in `codeCommitUserPublicKey`
+* `environmentVariables` - An array of environment variables passed into AWS CodeBuild.
+    e.g.:
+    ```json
+      {
+        "name": "FOO",
+        "value": "BAR"
+      }
+    ```
+* `stages` - the ordered list of stages where Code Pipeline will deploy the application.
```
Comment (nigelhanlon) [inline docs/config/README.md:112]:
```suggestion
* `stages` - An ordered list of stages where Code Pipeline will deploy the application.
```

File: docs/config/README.md [hunk 5]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
+Stage is described by 3 properties:
+    * `target` - name of the account used as a target account for the application deployment.
```
Comment (nigelhanlon) [inline docs/config/README.md:114]:
```suggestion
    * `target` - Name of the account used as a target for the application deployment (e.g staging)
```

File: docs/config/README.md [hunk 6]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
+    * `withDomain` - boolean that specifies if application supports custom domain __NB domain usage to follow in upcoming releases.__.
```
Comment (nigelhanlon) [inline docs/config/README.md:115]:
```suggestion
    * `withDomain` - A boolean that specifies if the application supports a custom domain. __NB domain usage to follow in upcoming releases.__.
```

File: docs/config/README.md [hunk 7]
```diff
@@ -72,52 +47,73 @@ Let's take a look at a sample config file.
...
+    * `requireManualApproval` - boolean that specifies if manual approval is needed in the pipeline before continuing deployment.
```
Comment (nigelhanlon) [inline docs/config/README.md:116]:
```suggestion
    * `requireManualApproval` - A boolean that specifies if manual approval is needed in the pipeline before continuing deployment.
```

**Comment by irelandm** [comment]
Please address this comment from Matteo: https://github.com/nearform/mira/issues/14#issuecomment-651708638

**Pull request merged**